kotest-assertions-arrow-fx-coroutines/src/commonTest/kotlin/io/kotest/assertions/arrow/fx/coroutines/ResourceExtensionRootTest.kt

```diff
@@ -0,0 +1,33 @@
+package io.kotest.assertions.arrow.fx.coroutines
+
+import io.kotest.core.extensions.install
+import io.kotest.core.spec.style.FunSpec
+
+class ResourceExtensionRootTest : FunSpec({
+  val sub = ResourceUnderTest()
+
+  install(ResourceExtension(sub.asResource(), LifecycleMode.Root)) {
+    configure()
+  }
```

Comment:
Can the extension be used to acquire two resources in a deterministic order?

Reply:
Yes, so `Resource` allows you to compose `Resource` in order, and their finalisers will be called in reverse order. So,

```
val res: Resource<String> = resource {
  val a = Resource(( "A".also(::println) }) { a, exitCase -> println("$a -> $exitCase") }.bind()
  Resource({ "$a => b") } { b, exitCase -> println("$b -> $exitCase") }.bind()
}
```

This will always print:

```kotlin
A
a => b
// tests
a => b - ExitCase.XXX
a - ExitCase.XXX
```

Cool think is you can use `parZip`, and Structured Concurrency safely inside the `resource` block.

If resources are independent of each other, they'll still run deterministically but depending on Kotest `beforeXXX` and `afterXXX`. So there order would then be:

```kotlin
A
a => b
// tests
a - ExitCase.XXX
a => b - ExitCase.XXX
```